en/getting-started/first-game/index.md

```diff
@@ -276,72 +579,230 @@ After reading and understanding the capabilities of the __Animation Editor__ cha
 
-### Making a Prefab
+So let's go ahead and create such a manager.
 
-For a node that needs to be generated repeatedly, it can be saved as a **Prefab (prefabricated)** resource. This means it can be used as a template when we dynamically generate other nodes of this same type.
+### Create GamManager
 
-> **Note**: before proceeding, please read the [Prefab Resources](../../asset/prefab.md) documentation.
+1. Right-click on the **Hierarchy** panel to create a new node named `GameManager`.
+2. Right-click in `assets/Scripts` folder on the **Assets** panel to create a TypeScript component named `GameManager`.
+3. Add the `GameManager` component to the GameManager node.
 
-It is necessary to make the basic element `cube` of the road into a __Prefab__, after which all three cubes in the __Scene__ can be deleted.
+### Create Prefabs
+
+For nodes that need to be duplicated, we can save them to a [Prefab](../../asset/prefab.md) asset as a template of dynamic assets.
+
+> Prefabs are a built-in asset type of the engine, the main purpose of which is to provide the possibility to instantiate some kind of node. Imagine if we had 1000 enemies in our game, creating them individually would take a lot of time, to solve this problem we could dynamically clone the prefabs 1000 times.
+
+Drag the elementary Cube node to the **Assets** panel to create a Prefab.
 
 ![create cube prefab](./images/create-cube-prefab.gif)
 
-### Adding the automatic road creation
+### Add Automatic Runway Creation Code
 
-A very long road is needed. The ideal method is to dynamically increase the length of the road, so that the `Player` can run forever. First, generate a fixed-length road with a length that is arbitrary. To do so, replace the code in the `GameManager` script with the following code:
+The player needs a long runway, and the ideal way is to dynamically increase the length of the runway so that it can run forever. Here for convenience, firstly we generate a fixed-length runway, the runway length can be defined by yourself. In addition, we can generate some pits on the runway and GameOver when the Player jumps to the pits.
 
-```ts
-import { _decorator, Component, Prefab, instantiate, Node, CCInteger} from 'cc';
-const { ccclass, property } = _decorator;
+First, we need to define some constants to indicate whether the current coordinates represent a pit or a rock. In Typescript, this can be done with enumerations.
+
+By using the keyword `enum` + the name of the enumeration we define the following enumeration, where `BT_NONE` represents a pit and `BT_STONE` represents a walkable surface.
 
+```typescript
 enum BlockType{
     BT_NONE,
-    BT_STONE,
+    BT_STONE, 
 };
+```
 
-@ccclass("GameManager")
-export class GameManager extends Component {
+Again, we need to define which prefabricated body or bodies the game manager will use to create the map.
```

Comment:
Again, we need to define which Prefab will be used by the GameManager to create the map.